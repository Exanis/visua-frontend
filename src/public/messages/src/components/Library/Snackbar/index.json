[
  {
    "id": "snackbar.close",
    "defaultMessage": "Close"
  }
]