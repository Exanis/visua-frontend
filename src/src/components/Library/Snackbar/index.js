import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import CloseIcon from '@material-ui/icons/Close';
import green from '@material-ui/core/colors/green';
import amber from '@material-ui/core/colors/amber';
import IconButton from '@material-ui/core/IconButton';
import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import WarningIcon from '@material-ui/icons/Warning';
import {withStyles} from '@material-ui/core/styles';

import {defineMessages, injectIntl} from 'react-intl';

// This part of the code directly came from mui examples

const messages = defineMessages({
    close: {
        'id': 'snackbar.close',
        'defaultMessage': 'Close',
    },
});

const variantIcon = {
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
};

const styles = theme => ({
    success: {
        backgroundColor: green[600],
    },
    error: {
        backgroundColor: theme.palette.error.dark,
    },
    info: {
        backgroundColor: theme.palette.primary.dark,
    },
    warning: {
        backgroundColor: amber[700],
    },
    icon: {
        fontSize: 20,
    },
    iconVariant: {
        opacity: 0.9,
        marginRight: theme.spacing.unit,
    },
    message: {
        display: 'flex',
        alignItems: 'center',
    },
});

export function MySnackbarContent(props) {
    const {classes, onClose, variant, intl, ...other} = props;
    const {formatMessage} = intl;
    const Icon = variantIcon[variant];

    return (
        <SnackbarContent
            className={classes[variant]}
            aria-describedby="client-snackbar"
            message={
                <span id="client-snackbar" className={classes.message}>
                    <Icon className={classNames(classes.icon, classes.iconVariant)}/>
                    {props.children}
                </span>
            }
            action={[
                <IconButton
                    key="close"
                    aria-label={formatMessage(messages.close)}
                    color="inherit"
                    className={classes.close}
                    onClick={onClose}
                >
                    <CloseIcon className={classes.icon}/>
                </IconButton>,
            ]}
            {...other}
        />
    );
}

MySnackbarContent.propTypes = {
    classes: PropTypes.object.isRequired,
    onClose: PropTypes.func,
    variant: PropTypes.oneOf(['success', 'warning', 'error', 'info']).isRequired,
};

const MySnackbarContentWrapper = injectIntl(withStyles(styles)(MySnackbarContent));

class VisuaSnackbar extends React.Component {
    state = {
        open: false
    };

    static getDerivedStateFromProps(props, state) {
        if (props.open) {
            return {open: true};
        }
        return state;
    }

    handleClose = (ev, reason) => {
        if (reason === 'clickaway') {
            return;
        }

        this.setState({open: false});
    };

    render() {
        return (
            <Snackbar
                anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'center',
                }}
                open={this.state.open}
                autoHideDuration={6000}
                onClose={this.handleClose}
            >
                <MySnackbarContentWrapper variant={this.props.variant} onClose={this.handleClose}>
                    {this.props.children}
                </MySnackbarContentWrapper>
            </Snackbar>
        );
    }
}

VisuaSnackbar.propTypes = {
    open: PropTypes.bool.isRequired,
    variant: PropTypes.oneOf(['success', 'warning', 'error', 'info']).isRequired
};

export default VisuaSnackbar;
