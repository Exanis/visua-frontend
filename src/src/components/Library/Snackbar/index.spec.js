import React from 'react';
import {shallow, mount} from 'enzyme';

import Snackbar, {MySnackbarContent} from './index';

describe('<Snackbar />', () => {
    const defaultProps = {
        open: true,
        variant: 'success'
    };

    it('should render Snackbar properly', () => {
        const rendererComponent = shallow(<Snackbar {...defaultProps} />);

        expect(rendererComponent).toMatchSnapshot();
    });

    it('should render Snackbar properly when hidden', () => {
        const props = {
            ...defaultProps,
            open: false
        };
        const rendererComponent = shallow(<Snackbar {...props} />);

        expect(rendererComponent).toMatchSnapshot();
    });

    it('should render Snackbar content properly', () => {
        const props = {
            classes: {},
            onClose: () => {
            },
            variant: 'success',
            intl: {
                formatMessage: () => 'test-intl'
            }
        };
        const rendererComponent = shallow(<MySnackbarContent {...props} />);

        expect(rendererComponent).toMatchSnapshot();
    });
});