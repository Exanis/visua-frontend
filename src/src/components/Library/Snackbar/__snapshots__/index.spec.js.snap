// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should render Snackbar content properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MySnackbarContent
    classes={Object {}}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    onClose={[Function]}
    variant="success"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "action": Array [
        <WithStyles(IconButton)
          aria-label="test-intl"
          className={undefined}
          color="inherit"
          onClick={[Function]}
>
          <pure(Close)
                    className={undefined}
          />
</WithStyles(IconButton)>,
      ],
      "aria-describedby": "client-snackbar",
      "className": undefined,
      "message": <span
        className={undefined}
        id="client-snackbar"
>
        <pure(CheckCircle)
                className=""
        />
</span>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "action": Array [
          <WithStyles(IconButton)
            aria-label="test-intl"
            className={undefined}
            color="inherit"
            onClick={[Function]}
>
            <pure(Close)
                        className={undefined}
            />
</WithStyles(IconButton)>,
        ],
        "aria-describedby": "client-snackbar",
        "className": undefined,
        "message": <span
          className={undefined}
          id="client-snackbar"
>
          <pure(CheckCircle)
                    className=""
          />
</span>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Snackbar /> should render Snackbar properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VisuaSnackbar
    open={true}
    variant="success"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "anchorOrigin": Object {
        "horizontal": "center",
        "vertical": "bottom",
      },
      "autoHideDuration": 6000,
      "children": <InjectIntl(WithStyles(MySnackbarContent))
        onClose={[Function]}
        variant="success"
/>,
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "onClose": [Function],
        "variant": "success",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "anchorOrigin": Object {
          "horizontal": "center",
          "vertical": "bottom",
        },
        "autoHideDuration": 6000,
        "children": <InjectIntl(WithStyles(MySnackbarContent))
          onClose={[Function]}
          variant="success"
/>,
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "onClose": [Function],
          "variant": "success",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Snackbar /> should render Snackbar properly when hidden 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VisuaSnackbar
    open={false}
    variant="success"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "anchorOrigin": Object {
        "horizontal": "center",
        "vertical": "bottom",
      },
      "autoHideDuration": 6000,
      "children": <InjectIntl(WithStyles(MySnackbarContent))
        onClose={[Function]}
        variant="success"
/>,
      "onClose": [Function],
      "open": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "onClose": [Function],
        "variant": "success",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "anchorOrigin": Object {
          "horizontal": "center",
          "vertical": "bottom",
        },
        "autoHideDuration": 6000,
        "children": <InjectIntl(WithStyles(MySnackbarContent))
          onClose={[Function]}
          variant="success"
/>,
        "onClose": [Function],
        "open": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "onClose": [Function],
          "variant": "success",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
